pymongo/bulk.py

```diff
@@ -290,33 +303,50 @@ def _execute_command(self, generator, write_concern, session,
                 sock_info.send_cluster_time(cmd, session, client)
                 sock_info.add_server_api(cmd)
                 ops = islice(run.ops, run.idx_offset, None)
+
                 # Run as many ops as possible in one command.
-                result, to_send = bwc.execute(cmd, ops, client)
-
-                # Retryable writeConcernErrors halt the execution of this run.
-                wce = result.get('writeConcernError', {})
-                if wce.get('code', 0) in _RETRYABLE_ERROR_CODES:
-                    # Synthesize the full bulk result without modifying the
-                    # current one because this write operation may be retried.
-                    full = copy.deepcopy(full_result)
-                    _merge_command(run, full, run.idx_offset, result)
-                    _raise_bulk_write_error(full)
-
-                _merge_command(run, full_result, run.idx_offset, result)
-                # We're no longer in a retry once a command succeeds.
-                self.retrying = False
-                self.started_retryable_write = False
-
-                if self.ordered and "writeErrors" in result:
-                    break
+                if write_concern.acknowledged:
+                    result, to_send = bwc.execute(cmd, ops, client)
+
+                    # Retryable writeConcernErrors halt the execution of this run.
+                    wce = result.get('writeConcernError', {})
+                    if wce.get('code', 0) in _RETRYABLE_ERROR_CODES:
+                        # Synthesize the full bulk result without modifying the
+                        # current one because this write operation may be retried.
+                        full = copy.deepcopy(full_result)
+                        _merge_command(run, full, run.idx_offset, result)
+                        _raise_bulk_write_error(full)
+
+                    _merge_command(run, full_result, run.idx_offset, result)
+
+                    # We're no longer in a retry once a command succeeds.
+                    self.retrying = False
+                    self.started_retryable_write = False
+
+                    if self.ordered and "writeErrors" in result:
+                        break
+                else:
+                    # Guard against unsupported unacknowledged writes.
```

Comment:
I think this guard should happen in the same place it was before (line 414) because we should raise ConfigurationError before executing any operation. To do that we might need two variables `self.hint_update` and `self.hint_delete`.

Reply:
Done